### ISSUE TITLE
Measuring the acoustic -> semantic -> text modeling difficulty

### ISSUE THREAD

**Issue by jpc** [collabora/WhisperSpeech#5]
The semantic tokens are supposed to be somewhere in between the acoustic tokens and text. Preferably they contain phonetic and lose all prosody/speaker information. Will Whisper embeddings perform this task successfully?

In the SPEAR TTS paper the semantic to acoustic task is relatively easy (they used a decoder-only model with 12 layers, about the size of Whisper Base) while the text to semantic task is very hard (T5-Large – 24 layer encoder + 24 layer decoder, the exact same size as Whisper Medium). Judging from our initial tests this will most likely not be the case for our Whisper-based semantic tokens. (https://github.com/collabora/spear-tts-pytorch/issues/4#issuecomment-1467757775)

To quantify this in some way we will try to train the Whisper decoders from scratch (using frozen encoder embeddings) on subsets of LibriLight (this is equivalent to the backtranslation task in the paper). For this task we know what great performance looks like (the pre-trained Whisper model) so we will be able to focus on dataset sizes and training hyperparameters.

**Comment by LorenzoBrugioni** [comment]
Hey @jpc , very interesting and great work ! How long do you think it will take to get some preliminary results to assess the potential of this approach?

**Comment by jpc** [comment]
I've added some more tasks (and updated the existing ones) that should help figure out where we are in the whole process. The most challenging task that remains is probably #9 but this is not linear – once we find and iron out the bugs the progress (and output quality) jumps discontinuously. Also #9 will most probably use the same overall architecture as #4 just different training data.

After a short break forced on my by the flu, we finally got some interesting sounding outputs yesterday.
